x86.c

```diff
@@ -1,12 +1,27 @@
 // x86 codegen
 
-#include "exe.c"
+#ifdef i386
+#include "elf.c"
+#endif
+
+#ifdef x86_64
 #include "elf.c"
+#endif
+
+#ifdef mac_os
+#include "mach-o.c"
+#endif
 
 #ifdef x86_64
-const int word_size = 8; // generating for x86-64
-#else
-const int word_size = 4; // generating for i386
+const int word_size = 8; // generating for x86-64 Linux
+#endif
```

Comment:
Nit: We can combine the `#ifdef` for `word_size`and for the `elf.c`/`mach-o.c` includes.

Reply:
Ok!